core/api_files.go

```diff
@@ -190,3 +190,33 @@ func (a *api) lsThreadFileTargetKeys(g *gin.Context) {
 
 	pbJSON(g, http.StatusOK, keys)
 }
+
+// getFileData godoc
+// @Summary File data at hash
+// @Description Returns raw data for file
+// @Tags files
+// @Produce application/octet-stream
+// @Param hash path string true "file hash"
+// @Success 200 {string} byte
+// @Failure 400 {string} string "Bad Request"
+// @Router /file/{hash}/data [get]
+func (a *api) getFileData(g *gin.Context) {
+	hash := g.Param("hash")
+
+	file, err := a.node.FileIndex(hash)
+	if err != nil {
+		g.String(http.StatusNotFound, err.Error())
+		return
+	}
+
+	contentLength := file.GetSize()
+	contentType := file.GetMedia()
+
+	reader, _, err := a.node.FileData(hash)
```

Comment:
doh, I just remembered this method returns the file index as it second param, may as well use it instead of loading it twice.